"ClientPayloadError: Response payload is not completed" with https://www.тв-програма.bg

```python
import aiohttp
import pytest


async def fetch():
    async with aiohttp.ClientSession() as session:
        remote_resp = await session.request("GET", "https://www.тв-програма.bg")
        print(remote_resp)
        await remote_resp.text()


@pytest.mark.asyncio
async def test():
    await fetch()
```

Results in:
```python-traceback
platform linux -- Python 3.7.3, pytest-5.0.1, py-1.8.0, pluggy-0.12.0 -- …/Vcs/aiohttp/.venv/bin/python
cachedir: .pytest_cache
rootdir: …/Vcs/aiohttp, inifile: pytest.ini
plugins: asyncio-0.10.0
collected 1 item

t-connection.py::test FAILED                                             [100%]

=================================== FAILURES ===================================
_____________________________________ test _____________________________________

    @pytest.mark.asyncio
    async def test():
>       await fetch()

t-connection.py:14:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
t-connection.py:9: in fetch
    await remote_resp.text()
aiohttp/client_reqrep.py:952: in text
    await self.read()
aiohttp/client_reqrep.py:916: in read
    self._body = await self.content.read()
aiohttp/streams.py:347: in read
    block = await self.readany()
aiohttp/streams.py:369: in readany
    await self._wait('readany')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <StreamReader e=ClientPayloadError('Response payload is not completed')>
func_name = 'readany'

    async def _wait(self, func_name: str) -> None:
        # StreamReader uses a future to link the protocol feed_data() method
        # to a read coroutine. Running two read coroutines at the same time
        # would have an unexpected behaviour. It would not possible to know
        # which coroutine would get the next data.
        if self._waiter is not None:
            raise RuntimeError('%s() called while another coroutine is '
                               'already waiting for incoming data' % func_name)

        waiter = self._waiter = self._loop.create_future()
        try:
            if self._timer:
                with self._timer:
>                   await waiter
E                   aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed

aiohttp/streams.py:297: ClientPayloadError
----------------------------- Captured stdout call -----------------------------
<ClientResponse(https://www.xn----8sbafg9clhjcp.bg) [200 OK]>
<CIMultiDictProxy('Date': 'Sun, 14 Jul 2019 18:32:22 GMT', 'Server': 'Apache', 'Expires': 'Thu, 19 Nov 1981 08:52:00 GMT', 'Cache-Control': 'no-store, no-cache, must-revalidate', 'Pragma': 'no-cache', 'Set-Cookie': 'PHPSESSID=XXX; path=/; domain=.xn----8sbafg9clhjcp.bg', 'Upgrade': 'h2', 'Connection': 'Upgrade', 'Vary': 'Accept-Encoding', 'Content-Encoding': 'gzip', 'Transfer-Encoding': 'chunked', 'Content-Type': 'text/html; charset=utf-8')>

----------------------------- Captured stderr call -----------------------------
Exception ignored in: <function _SSLProtocolTransport.__del__ at 0x7fdc87bf32f0>
Traceback (most recent call last):
  File "/usr/lib/python3.7/asyncio/sslproto.py", line 322, in __del__
    source=self)
ResourceWarning: unclosed transport <asyncio.sslproto._SSLProtocolTransport object at 0x7fdc87053f98>
=========================== 1 failed in 5.39 seconds ===========================
```

URL: https://www.тв-програма.bg

It works fine using a browser or curl.

There are not response headers related to content length:
```http
HTTP/2 200
date: Sun, 14 Jul 2019 18:33:24 GMT
server: Apache
expires: Thu, 19 Nov 1981 08:52:00 GMT
cache-control: no-store, no-cache, must-revalidate
pragma: no-cache
set-cookie: PHPSESSID=XXX; path=/; domain=.xn----8sbafg9clhjcp.bg
content-type: text/html; charset=utf-8
```

Using aiohttp master (95ead73).

--- a/CHANGES/8720.bugfix.rst
+++ b/CHANGES/8720.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed an edge case in the Python parser when chunk separators happen to align with network chunks -- by :user:`Dreamsorcerer`.

--- a/aiohttp/http_parser.py
+++ b/aiohttp/http_parser.py
@@ -845,13 +845,13 @@ class HttpPayloadParser:
                     if self._chunk_size:
                         return False, b""
                     chunk = chunk[required:]
-                    if self._lax and chunk.startswith(b"\r"):
-                        chunk = chunk[1:]
                     self._chunk = ChunkState.PARSE_CHUNKED_CHUNK_EOF
                     self.payload.end_http_chunk_receiving()
 
                 # toss the CRLF at the end of the chunk
                 if self._chunk == ChunkState.PARSE_CHUNKED_CHUNK_EOF:
+                    if self._lax and chunk.startswith(b"\r"):
+                        chunk = chunk[1:]
                     if chunk[: len(SEP)] == SEP:
                         chunk = chunk[len(SEP) :]
                         self._chunk = ChunkState.PARSE_CHUNKED_SIZE

--- a/tests/test_http_parser.py
+++ b/tests/test_http_parser.py
@@ -1461,6 +1461,30 @@ def test_parse_chunked_payload_empty_body_than_another_chunked(
     assert b"second" == b"".join(d for d in payload._buffer)
 
 
+async def test_parse_chunked_payload_split_chunks(response: HttpResponseParser) -> None:
+    network_chunks = (
+        b"HTTP/1.1 200 OK\r\n"
+        b"Transfer-Encoding: chunked\r\n\r\n"
+        b"5\r\nfi",
+        b"rst",
+        # This simulates a bug in lax mode caused when the \r\n separator, before the
+        # next HTTP chunk, appears at the start of the next network chunk.
+        b"\r\n",
+        b"6",
+        b"\r",
+        b"\n",
+        b"second\r",
+        b"\n0\r\n\r\n"
+    )
+    reader = response.feed_data(network_chunks[0])[0][0][1]
+    for c in network_chunks[1:]:
+        response.feed_data(c)
+
+    assert response.feed_eof() is None
+    assert reader.is_eof()
+    assert await reader.read() == b"firstsecond"
+
+
 def test_partial_url(parser: HttpRequestParser) -> None:
     messages, upgrade, tail = parser.feed_data(b"GET /te")
     assert len(messages) == 0

--- a/tests/test_http_parser.py
+++ b/tests/test_http_parser.py
@@ -1463,9 +1463,7 @@ def test_parse_chunked_payload_empty_body_than_another_chunked(
 
 async def test_parse_chunked_payload_split_chunks(response: HttpResponseParser) -> None:
     network_chunks = (
-        b"HTTP/1.1 200 OK\r\n"
-        b"Transfer-Encoding: chunked\r\n\r\n"
-        b"5\r\nfi",
+        b"HTTP/1.1 200 OK\r\n" b"Transfer-Encoding: chunked\r\n\r\n" b"5\r\nfi",
         b"rst",
         # This simulates a bug in lax mode caused when the \r\n separator, before the
         # next HTTP chunk, appears at the start of the next network chunk.
@@ -1474,7 +1472,7 @@ async def test_parse_chunked_payload_split_chunks(response: HttpResponseParser) 
         b"\r",
         b"\n",
         b"second\r",
-        b"\n0\r\n\r\n"
+        b"\n0\r\n\r\n",
     )
     reader = response.feed_data(network_chunks[0])[0][0][1]
     for c in network_chunks[1:]:

--- a/tests/test_http_parser.py
+++ b/tests/test_http_parser.py
@@ -1463,7 +1463,9 @@ def test_parse_chunked_payload_empty_body_than_another_chunked(
 
 async def test_parse_chunked_payload_split_chunks(response: HttpResponseParser) -> None:
     network_chunks = (
-        b"HTTP/1.1 200 OK\r\n" b"Transfer-Encoding: chunked\r\n\r\n" b"5\r\nfi",
+        b"HTTP/1.1 200 OK\r\n",
+        b"Transfer-Encoding: chunked\r\n\r\n",
+        b"5\r\nfi",
         b"rst",
         # This simulates a bug in lax mode caused when the \r\n separator, before the
         # next HTTP chunk, appears at the start of the next network chunk.

--- a/tests/test_http_parser.py
+++ b/tests/test_http_parser.py
@@ -1463,8 +1463,7 @@ def test_parse_chunked_payload_empty_body_than_another_chunked(
 
 async def test_parse_chunked_payload_split_chunks(response: HttpResponseParser) -> None:
     network_chunks = (
-        b"HTTP/1.1 200 OK\r\n",
-        b"Transfer-Encoding: chunked\r\n\r\n",
+        b"HTTP/1.1 200 OK\r\nTransfer-Encoding: chunked\r\n\r\n",
         b"5\r\nfi",
         b"rst",
         # This simulates a bug in lax mode caused when the \r\n separator, before the

